packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -222,21 +236,75 @@ async function selectPlatformAndExecute(
     const message = isAndroid
       ? 'force_lightning_lwc_android_target_remembered'
       : 'force_lightning_lwc_ios_target_remembered';
-    placeholderText = nls.localize(message, lastTarget);
+    createDevicePlaceholderText = nls.localize(message, lastTarget);
     target = lastTarget;
   }
-  const targetName = await vscode.window.showInputBox({
-    placeHolder: placeholderText
-  });
 
-  if (targetName === undefined) {
-    vscode.window.showInformationMessage(
-      isAndroid
-        ? nls.localize('force_lightning_lwc_android_device_cancelled')
-        : nls.localize('force_lightning_lwc_ios_device_cancelled')
-    );
-    return;
+  const deviceListOutput = new CommandOutput();
+  const deviceListCommand = new SfdxCommandBuilder()
+    .withArg(sfdxDeviceListCommand)
+    .withFlag('-p', platformSelection.platformName)
+    .withJson()
+    .build();
+
+  const mobileCancellationTokenSource = new vscode.CancellationTokenSource();
+  const mobileCancellationToken = mobileCancellationTokenSource.token;
+  const deviceListExecutor = new CliCommandExecutor(deviceListCommand, {});
+  const deviceListExecution = deviceListExecutor.execute(
+    mobileCancellationToken
+  );
+
+  const options: vscode.QuickPickItem[] = [];
+  let targetName: string | undefined;
+  try {
+    const result = await deviceListOutput.getCmdResult(deviceListExecution);
+    const jsonString = result.substring(result.indexOf('{'));
+    const json = JSON.parse(jsonString);
+    const devices = json.result as any[];
+
+    devices.forEach(device => {
+      const label = isAndroid ? device.displayName : device.name;
+      const detail = isAndroid
+        ? `${device.target}, ${device.api}`
+        : device.runtimeId;
+      options.push({ label, detail });
+    });
+  } catch (e) {
+    // If device enumeration failes for any reason, we silently fail
+    // and proceed with an empty list of devices.
   }
+
+  // if there are any devices available, show a pick list.
```

Comment:
Not for this PR, but I would like to use the "remembered device" concept to show the last several used devices at the top of the list.

Reply:
That's a good idea. Feel free to log a feature request on our repo for this and I can pick that up next (or whenever I get to it).